lib/coveralls.rb

```diff
@@ -44,6 +44,7 @@ def setup!
       begin
         require 'simplecov'
         @adapter = :simplecov if defined?(::SimpleCov)
+      rescue LoadError # rubocop:disable Lint/HandleExceptions
```

Comment:
Is there a reason to have the `LoadError` swallowed silently, while the `StandardError`'s message gets printed to STDOUT?
```
      rescue LoadError, StandardError => e # rubocop:disable Lint/HandleExceptions
        # TODO: Add error action
        puts e.message
```

Reply:
@pboling 

`webmock` and `vcr` are not requirements of this gem. I think that when a `LoadError` occurs there's not an error in that piece of code.

Example: my application uses `webmock`, so `coveralls-ruby` should add `coveralls.io` to `webmock`'s allow list before sending results. 

If `vcr` is not found, it is fine.

In a such an application, if you print the error message for `LoadError`, you will see:

```
Finished in 0.17949 seconds (files took 2.95 seconds to load)
1 example, 0 failures

Randomized with seed 56372

cannot load such file -- vcr
[Coveralls] Submitting to https://coveralls.io/api/v1
```

I think that this is not something that we should see.

If we really want to see something, we should output "[Coveralls] Allowing connections to `coveralls.io` through Webmock" or "[Coveralls] Allowing connections to `coveralls.io` through VCR" but when the require succeed

Any suggestion?

Reply:
> Any suggestion? 

```ruby
expect { subject }.to output('cannot load such file -- vcr').to_stderr

# for other tests

allow($stderr).to receive(:puts).with('cannot load such file -- vcr) # I'm not sure about method
```

Reply:
@AlexWayfer

The problem is in the application that uses coveralls, not in coveralls or its tests.

When I run tests in my application, If `coveralls` rescues from LoadError with an error message, I see `'cannot load such file -- vcr'` on my console.

That message doesn't make any sense in my application, because I'm not using `vcr` and `vcr` is not a dependency of `coveralls`

Reply:
OK, I got it. If it non-required require in any application — then yes, output is not necessary. 

Or this require can be outside and described in docs. And then `const_defined?` or something like that. So, if a project wants to use `vcr` — it should require it previously.

> I'm available to switch back to the old code that checks if vcr or webmock are already defined. Ref: lemurheavy#103

Oh, this sounds just like I'm writing!

> Sorry, I'm not an English native speaker

It's OK, many of us are not English native.

Reply:
This still doesn't make sense to me.
The entire block that is being handled by these rescues is:
```
      begin
        require 'simplecov'
        @adapter = :simplecov if defined?(::SimpleCov)
```
Therefore the *only* thing that could fail and raise a `LoadError` is the attempt to require simplecov, which is done by coveralls.

vcr is not in any way relevant, and if it seems to be I think something else is very wrong.

Reply:
...btw, I've just realized that this comment is about the rescue from LoadError related to simplecov 🤦‍♂ 🤦‍♂ 🤦‍♂

Reply:
LOL.  Groks are hard ;)  Cheers!